spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/source/SparkScanBuilder.java

```diff
@@ -158,6 +179,129 @@ public Filter[] pushedFilters() {
+          return false;
+        }
+        if (!aggregateEvaluator.update(task.file())) {
+          return false;
+        }
+      }
+    } catch (IOException e) {
+      LOG.info("Skipped aggregate pushdown: " + e);
+      return false;
+    }
+
+    pushedAggregateSchema =
+        SparkSchemaUtil.convert(new Schema(aggregateEvaluator.resultType().fields()));
+    this.pushedAggregateRows = new InternalRow[1];
+    StructLike structLike = aggregateEvaluator.result();
+    this.pushedAggregateRows[0] =
+        new StructInternalRow(aggregateEvaluator.resultType()).setStruct(structLike);
+    return true;
+  }
+
+  private boolean canPushDownAggregation(Aggregation aggregation) {
+    if (!(table instanceof BaseTable)) {
+      return false;
+    }
+
+    if (!readConf.aggregatePushDownEnabled()) {
+      return false;
+    }
+
+    // If group by expression is the same as the partition, the statistics information can still
+    // be used to calculate min/max/count, will enable aggregate push down in next phase.
+    // TODO: enable aggregate push down for partition col group by expression
+    if (aggregation.groupByExpressions().length > 0) {
+      LOG.info("Skipped aggregate pushdown: group by aggregation push down is not supported.");
+      return false;
+    }
+
+    return true;
+  }
+
+  private Snapshot readSnapshot() {
+    Snapshot snapshot;
+    if (readConf.snapshotId() != null) {
+      snapshot = table.snapshot(readConf.snapshotId());
+    } else {
+      snapshot = table.currentSnapshot();
+    }
+
+    return snapshot;
+  }
+
+  private boolean metricsModeSupportsAggregatePushDown(List<BoundAggregate<?, ?>> aggregates) {
+    MetricsConfig config = MetricsConfig.forTable(table);
+    for (BoundAggregate aggregate : aggregates) {
+      String colName = aggregate.columnName();
+      if (!colName.equals("*")) {
+        MetricsModes.MetricsMode mode = config.columnMode(colName);
+        if (mode instanceof MetricsModes.None) {
+          LOG.info("Skipped aggregate pushdown: MetricsModes.None");
+          return false;
+        } else if (mode instanceof MetricsModes.Counts) {
+          if (aggregate.op() == Expression.Operation.MAX
+              || aggregate.op() == Expression.Operation.MIN) {
+            LOG.info(
+                "Skipped aggregate pushdown: MetricsModes.Counts, but the aggregate is "
+                    + aggregate);
+            return false;
+          }
+        } else if (mode instanceof MetricsModes.Truncate) {
+          // lower_bounds and upper_bounds may be truncated, so disable push down
+          if (aggregate.type().typeId() == Type.TypeID.STRING) {
+            if (aggregate.op() == Expression.Operation.MAX
+                || aggregate.op() == Expression.Operation.MIN) {
+              LOG.info(
+                  "Skipped aggregate pushdown: MetricsModes.Truncate, but the aggregate is MIN/MAX on a String column.");
```

Comment:
How about `"Skipped aggregate pushdown: Cannot produce min or max from truncated values for column %s"`?